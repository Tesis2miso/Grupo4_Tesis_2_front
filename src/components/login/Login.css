.login {
    background-color: rgb(26, 53, 235);
    min-height: 100vh;
    display: flex;
    flex-direction: column;
    align-items: center;
    justify-content: center;
    font-size: calc(10px + 2vmin);
    color: white;
  }


.label_error
{
  font-size:small;
  color: red;
}

a {
  font-size: 20px;
  color: rgb(226, 189, 247);
}
a:visited {
  color: rgb(226, 189, 247);
}
a:hover, a:active, a:focus {
  color: rgb(156, 111, 111);
}