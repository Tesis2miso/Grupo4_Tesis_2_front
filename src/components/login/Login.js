import { useState, Fragment } from 'react';
import axios from "axios";
import './Login.css';

function Login(props) {

    const [loginForm, setloginForm] = useState({
        email: "",
        password: ""
    })
    const [error, setError] = useState();

    function logMeIn(event) {
        axios({
            method: "POST",
            url: "/specialist/login",
            data: {
                email: loginForm.email,
                password: loginForm.password
            }
        })
            .then((response) => {
                props.setToken(response.data.access_token)
                localStorage.setItem("userName", response.data.username.username)
            }).catch((error) => {
                if (error.response) {
                    console.log(error.response)
                    console.log(error.response.status)
                    console.log(error.response.headers)
                }
                setError('Invalid Username or Password')
            })

        setloginForm(({
            email: "",
            password: ""
        }))

        event.preventDefault()
    }

    function handleChange(event) {
        const { value, name } = event.target
        setloginForm(prevNote => ({
            ...prevNote, [name]: value
        })
        )
    }

    document.addEventListener('DOMContentLoaded', () => {
        const miInput = document.getElementById('password');

        miInput.addEventListener('keyup', function (event) {
            if (event.getModifierState('CapsLock')) {
                setError('Alert: Caps are on');
            } else {
                setError('');
            }
        });
    });

    return (
        <Fragment>
            <form className="login">
                <h1>DermoApp</h1>
                <br></br>
                <input id="username" onChange={handleChange}
                    type="email"
                    text={loginForm.email}
                    name="email"
                    placeholder="Email"
                    value={loginForm.email} />
                <input id="password" onChange={handleChange}
                    type="password"
                    text={loginForm.password}
                    name="password"
                    placeholder="Password"
                    value={loginForm.password} />
                <div><label className='label_error'>{error}</label></div>
                <table class="tg">
                    <thead>
                        <tr>
                            <td class="tg-0pky"><button id="submitbtn1" onClick={logMeIn}>Submit</button></td>
                        </tr>
                        <tr>
                            <td class="tg-0pky"><a href='#'>Forgot password?</a></td>
                        </tr>
                    </thead>
                </table>
            </form>
        </Fragment>
    );
}

export default Login;