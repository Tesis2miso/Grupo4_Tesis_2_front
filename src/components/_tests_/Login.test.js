import '@testing-library/jest-dom';
import { shallow } from 'enzyme';
import Login from '../login/Login';


describe('Tests_Login', () => {
  let wrapper = shallow(<Login />);
  beforeEach(() => {
    wrapper = shallow(<Login />);
  });

  test('show_Login', () => {
    expect(wrapper).toMatchSnapshot();
  });

  test('show_Login_elements', () => {
    expect(wrapper.find('h1').text()).toEqual('DermoApp');
    expect(wrapper.find('#username')).toHaveLength(1);
    expect(wrapper.find('#password')).toHaveLength(1);
    expect(wrapper.find('#submitbtn1').text()).toEqual('Submit');
  });

  test('show_empty_inputs', () => {
    expect(wrapper.find('#username').text()).toEqual("");
    expect(wrapper.find('#password').text()).toEqual("");
  });

})