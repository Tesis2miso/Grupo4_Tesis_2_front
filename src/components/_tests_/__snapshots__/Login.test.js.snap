// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests_Login show_Login 1`] = `
<Fragment>
  <form
    className="login"
  >
    <h1>
      DermoApp
    </h1>
    <br />
    <input
      id="username"
      name="email"
      onChange={[Function]}
      placeholder="Email"
      text=""
      type="email"
      value=""
    />
    <input
      id="password"
      name="password"
      onChange={[Function]}
      placeholder="Password"
      text=""
      type="password"
      value=""
    />
    <div>
      <label
        className="label_error"
      />
    </div>
    <table
      class="tg"
    >
      <thead>
        <tr>
          <td
            class="tg-0pky"
          >
            <button
              id="submitbtn1"
              onClick={[Function]}
            >
              Submit
            </button>
          </td>
        </tr>
        <tr>
          <td
            class="tg-0pky"
          >
            <a
              href="#"
            >
              Forgot password?
            </a>
          </td>
        </tr>
      </thead>
    </table>
  </form>
</Fragment>
`;
